sys/arduino/base.cpp

```diff
@@ -58,3 +61,11 @@ void delay(unsigned long msec)
 {
     xtimer_usleep(1000 * msec);
 }
+
+#ifdef ADC_NUMOF
+int analogRead(int pin)
+{
+    adc_init((adc_t)pin);
+    return adc_sample((adc_t)pin, (adc_res_t)0);
```

Comment:
don't think the simple cast of `pin` to `adc_t` will suffice here. IMHO this functions needs 
- check, if the given pin is a analog pin
- map the analog pin number to a RIOT ADC_LINE, possibly something like `ADC_LINE(pin - SOME_OFFSET?)`